### ISSUE TITLE
Installation on MacOS is confusing to new/inexperienced users, does it have to be this difficult?

### ISSUE THREAD

**Issue by BPBaker** [reaper-oss/sws#1505]
Hello--I'm sure this is an issue you hear about frequently but I'm hoping to explain why I think it's worth addressing, if possible!

I'm teaching a series of online Zoom tutorials training radio/podcast producers how to use REAPER (and generally evangelizing for REAPER) and installing SWS is a critical part of the system I'm teaching.

I'm getting *hundreds* of people per Zoom lesson (using Mac and PC, with a wide range of tech skills/comfort, age, etc.), and _installing SWS on the Mac side is one of the real pain points in my class._

On the PC side, everything goes smoothly with the .exe-based installers. But I run into SO many issues with SWS installation on the Mac side--particularly with people who have less tech experience. And even though I walk users through the whole process step by step, **these issues really, truly, derail the class and prevent me from getting people on board with REAPER**--making the whole DAW seem more daunting than it really is.

Even though I explain this all in my class, sometimes the issue is just confusion about finding the UserPlugins folder. Sometimes it's about the broken link on the SWS .dmg. And sometimes it's the issue of notarization on Catalina/Big Sur, having to go into System Preferences and manually allowing something from an unidentified developer. Sometimes it's an issue of *experienced* users moving too fast rather than listening to me! ;-)

I'm sure there is a reason the current Mac SWS installation is the way it is, but I'm writing it ask: **is there a way to simplify installation process on the Mac either with creating an automated .pkg and/or fixing the link to "UserPlugins" on the SWS .dmg** (which seems to have been broken for several years now)?

Thanks for your work maintaining SWS and let me know if there's any info I can provide that might be helpful in helping address this issue!

- Brendan

**Comment by cfillion** [comment]
The broken link is unfixable because the home directory path varies from user to user. Replacing it with an install script in the DMG should be possible (unless notarization blocks it?). OSARA uses that.

There is no practical solution notarization (aka. Apple forcing developers to submit their software to them for holy approval). It's troublesome and costs money. I've heard reports of the System Preferences method not working for everybody on Big Sur. A simple way to reliably do it is by copy/pasting this into Terminal.app:

```
xattr -d com.apple.quarantine ~/Library/'Application Support'/REAPER/UserPlugins/reaper_sws-*.dylib
```

**Comment by BPBaker** [comment]
Thanks for the quick reply and understood RE Apple notarization. (I suspected that would probably be the case. But thankfully  for the most part I've been able to walk users through "allowing" it via Security Preferences.)

But, yeah, having an automated install script built into the SWS .dmg that would work with the standard/default REAPER installation would be a HUGE help and make SWS much more accessible to non-power users. Basically anything that's easily clickable or drag/droppable from the DMG window itself--once the installation involves having to navigate manually to the REAPER resource directory, that's where most of the user error seems to occur in my experience. (And I suspect most of those non-power users would also find terminal-based installation "too complicated," too.) Again, the way you do this on the PC side is a breeze and I haven't run into any user error there so far.

**Comment by cfillion** [comment]
I just found out that SWS used to have a install script in the macOS disk image before v2.7.1. All it did was run a few `mkdir`/`cp` commands (so it didn't support portable installs). https://forum.cockos.com/showpost.php?p=1518816

**Comment by X-Raym** [comment]
Note that UserPlugins folder is easily accessible via REAPER Option menu -> Show user resource folder in finder. Then it is just a cmick and drag of the SWS file and it is ready.

**Comment by swstim** [comment]
I will look into (ie purchase) the right tools to get the releases signed.

**Comment by BPBaker** [comment]
> Note that UserPlugins folder is easily accessible via REAPER Option menu -> Show user resource folder in finder. Then it is just a cmick and drag of the SWS file and it is ready.

Yeah, this is how I've been instructing people as it's the most straightforward way to instal SWS on the Mac as is. But you'd be surprised how even this extra step makes it significantly harder when you're trying to teach 100+ people how install it via Zoom. ;-) (Meanwhile it's a non-issue on the PC side.) Point being: there must be a better way! Sounds like y'all are looking into it, though! :-)  Thanks again and I appreciate your work!



> I will look into (ie purchase) the right tools to get the releases signed.

Amazing!

**Comment by BPBaker** [comment]
FYI, I’m doing several more of these Zoom courses starting Wednesday where I’ll have new users install SWS. If there’s a beta install script that you’d like to test out with my group, let me know!

**Comment by BPBaker** [comment]
Just checking to see if by any chance there's a beta/interim build available that helps with this issue? I'm teaching Reaper+SWS to some young students next week--some using Macs--and I'd would love to make installation easier on them if possible. Thanks!

**Comment by BPBaker** [comment]
Hello, @cfillion @swstim @nofishonfriday. Following up on this old thread to ask if any more can be done at this point to simplify or streamline the installation on MacOS. I'm currently teaching REAPER to radio/podcast journalism students at NYU, and SWS installation on Macs continues to be a major sticking point for many students. Some people have problems getting to the UserPlugins folder. Others are derailed by the Security approval process. 

Would it be possible to create an MacOS installer script that would automate 1) copying the reaper_sws-*.dylib file to appropriate MacOS *user* library (rather than system library, as seems to still be what the .dmg points toward) and 2) quarantine (or somehow pre-approve) the .dylib? And/or could you get the tools to notarize/sign SWS to avoid the security issue?

I know the manual installation doesn't seem like that big to deal to people with tech savvy, but this has continues to be a significant hurdle for my students, many of whom are first-time REAPER users. Anything that would help streamline this process and make it more automated (like the PC installer for SWS) would be immensely helpful!
